www/js/lib/settingsStore.js

```diff
@@ -0,0 +1,129 @@
+'use strict';
+define([], function () {
+  /*\
+  |*|
+  |*|  :: settingsStore.js ::
+  |*|
+  |*|  A reader/writer framework for cookies or localStorage with full unicode support based on Mozilla cookies framework.
+  |*|
+  |*|  Revision #1 - September 4, 2014
+  |*|
+  |*|  https://developer.mozilla.org/en-US/docs/Web/API/document.cookie
+  |*|  https://developer.mozilla.org/User:fusionchess
+  |*|
+  |*|  This framework is released under the GNU Public License, version 3 or later.
+  |*|  http://www.gnu.org/licenses/gpl-3.0-standalone.html
+  |*|
+  |*|  Syntaxes:
+  |*|
+  |*|  * settingsStore.setItem(name, value[, end[, path[, domain[, secure]]]])
+  |*|  * settingsStore.getItem(name)
+  |*|  * settingsStore.removeItem(name[, path[, domain]])
+  |*|  * settingsStore.hasItem(name)
+  |*|
+  \*/
+
+  var storeType = testStorageSupport();
+
+  // Tests for localStorage or cookie support
+  function testStorageSupport() {
+    // DEV: In FF extensions, cookies are blocked since at least FF 68.6 but possibly since FF 55 [kiwix-js #612]
+    var type = 'none';
+    // First test for localStorage API support
+    var localStorageTest;
+    try {
+      localStorageTest = 'localStorage' in window && window['localStorage'] !== null;
+      // DEV: Above test returns true in IE11 running from file:// protocol, but attempting to write a key to
+      // localStorage causes an exception; so to test fully, we must now attempt to write and remove a test key
+      if (localStorageTest) {
+        localStorage.setItem('tempKiwixStorageTest', '');
+        localStorage.removeItem('tempKiwixStorageTest');
+      }
+    } catch (e) {
+      localStorageTest = false;
+    }
+    // Now test for document.cookie API support
+    document.cookie = 'tempKiwixCookieTest=working;expires=Fri, 31 Dec 9999 23:59:59 GMT';
+    var kiwixCookieTest = /tempKiwixCookieTest=working/.test(document.cookie);
+    // Remove test value by expiring the key
+    document.cookie = 'tempKiwixCookieTest=;expires=Thu, 01 Jan 1970 00:00:00 GMT';
+    if (kiwixCookieTest) type = 'cookie';
+    // Prefer localStorage if supported due to some platforms removing cookies once the session ends in some contexts
+    if (localStorageTest) type = 'local_storage';
+    // Update API panel with API name
+    var settingsStoreStatusDiv = document.getElementById('settingsStoreStatus');
```

Comment:
The settingsStore.js file must not access the DOM. It's not its responsibility : it should only deal with preferences storage.
What could be done instead is expose a function here, that returns the storeType variable. This function could be called by app.js in the existing refreshAPIStatus function, where it would display what we want.

Reply:
Ah OK! No problem. Will put the code into app.js and get the value from settingsStore.js.

Reply:
Done.